Bio/AlignIO/MafIO.py

```diff
@@ -0,0 +1,774 @@
+
+                annotations = []
+                records = []
+            else:
+                raise ValueError("Error parsing alignment - unexpected line:\n%s" % (line,))
+        elif STARTSWITH(line, "a"):
+            # start a bundle of records
+            in_a_bundle = True
+            annot_strings = SPLIT(STRIP(line))[1:]
+            if len(annot_strings) != line.count("="):
+                raise ValueError("Error parsing alignment - invalid key in 'a' line")
+
+            annotations = dict([SPLIT(a_string, "=") for a_string in annot_strings])
+        elif STARTSWITH(line, "#"):
+            # ignore comments
+            pass
+        elif not line:
+            break
+
+
+class MafIndex(object):
+    """This is used as an index for a MAF file.
+
+    The index is a sqlite3 database that is build upon creation of the object
+    if necessary, and queried when methods *search* or *get_spliced* are
+    used."""
+
+    def __init__(self, sqlite_file, maf_file, target_seqname):
+        """Indexes or loads the index of a MAF file"""
+
+        import os
+
+        try:
+            from sqlite3 import dbapi2 as _sqlite
+        except ImportError:
+            from Bio import MissingPythonDependencyError
+            raise MissingPythonDependencyError(
+                "Requires sqlite3, which is included in Python 2.5+")
+
+        self._target_seqname = target_seqname
+        self._maf_file = maf_file
+
+        # make sure maf_file exists, then open it up
+        if os.path.isfile(self._maf_file):
+            self._maf_fp = open(self._maf_file, "r")
+        else:
+            raise ValueError("Error opening %s -- file not found" % (self._maf_file,))
+
+        # if sqlite_file exists, use the existing db, otherwise index the file
+        if os.path.isfile(sqlite_file):
+            self._db_con = _sqlite.connect(sqlite_file)
+            self._record_count = self.__check_existing_db()
+        else:
+            self._db_con = _sqlite.connect(sqlite_file)
+            self._record_count = self.__make_new_index()
+
+        # lastly, setup a MafIterator pointing at the open maf_file
+        self._mafiter = MafIterator(self._maf_fp)
+
+    def __check_existing_db(self):
+        """Basic sanity checks upon loading an existing index"""
+
+        import os
+        from sqlite3 import OperationalError as _OperationalError
+        from sqlite3 import DatabaseError as _DatabaseError
+
+        try:
+            idx_version = int(self._db_con.execute(
+                "SELECT value FROM meta_data WHERE key = 'version'").fetchone()[0])
+            if idx_version != MAFINDEX_VERSION:
+                raise ValueError(
+                    "Index version (%s) incompatible with this version of MafIndex" % (
+                        idx_version,))
+
+            filename = self._db_con.execute(
+                "SELECT value FROM meta_data WHERE key = 'filename'").fetchone()[0]
+            if filename != self._maf_file:
```

Comment:
Should this remove the file path and compare only the base name? See also changes to the SeqIO indexing for relatives paths made pull request #381 included in Biopython 1.65?